vulture/lines.py

```diff
@@ -23,27 +25,26 @@
     })
 
 
-# Reinventing wheel here to keep vulture lightweight.
-def memoize(func):
-    _cache = {}
+def _get_range(entry):
+    if isinstance(entry, ENTRY_POINTS):
+        last_body = entry.body[-1]
+        while isinstance(last_body, tuple(TRAVERSABLE_FIELDS.keys())):
+            fields = TRAVERSABLE_FIELDS[last_body.__class__]
+            for field in reversed(fields):
+                children = getattr(last_body, field)
+                if children:
+                    break
+            last_body = children[-1]  # The last child.
 
-    @wraps(func)
-    def wrapped(*args):
-        try:
-            return _cache[args]
-        except KeyError:
-            _cache[args] = result = func(*args)
-            return result
-    return wrapped
+        last_line = last_body.lineno
+        SourceRange = namedtuple("SourceRange",
+                                 ['name', 'start_line', 'last_line'])
+        source_range = SourceRange(entry.name, entry.lineno, last_line)
+        yield source_range
 
 
-@memoize
-def estimate_lines(node):
-    """
-    Recursively count child AST nodes under `node`. It is an approximation
-    of the amount of code belonging to the node, which is useful for
-    sorting the list of unused code that a developer might want to remove.
-    """
-    return 1 + sum(estimate_lines(child)
-                   for field in TRAVERSABLE_FIELDS.get(node.__class__, ())
-                   for child in getattr(node, field))
+def count_lines(code):
+    span = 1
+    for source_range in _get_range(code):
+        span = (1 + source_range.last_line - source_range.start_line)
+    return span
```

Comment:
I think this can just be 
    
    def count_lines(node):
        return get_last_line_number(node) - node.lineno + 1